File: climbing-stairs/climbing-stairs.java
class Solution {
    HashMap<Integer,Integer>map = new HashMap<>();
    public int climbStairs(int n) {
        return climb(n,0);
    }
    int climb(int n, int i){
        if(i==n)return 1;
        if(i>n)return 0;
        if(map.containsKey(i))return map.get(i);
        int sum = climb(n,i+1)+climb(n,i+2);
        map.put(i,sum);
        return map.get(i);
    }
}

File: 70-climbing-stairs/climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        stairs = [0]*(n+1)
        stairs[0] = 1
        for i in range(1,n+1):
            stairs[i]+=stairs[i-1]
            if i-2>=0:
                stairs[i]+=stairs[i-2]
        return stairs[n]